flake.nix

```diff
@@ -6,29 +6,25 @@
 
   outputs = inputs: inputs.flake-utils.lib.eachSystem [ "x86_64-linux" "x86_64-darwin" ] (system:
     let
-      mob-overlay = self: super: {
-        mob = super.mob.overrideAttrs (old: rec {
-          version = "4.0.0";
-
-          src = super.fetchFromGitHub {
-            owner = "remotemobprogramming";
-            repo = old.pname;
-            rev = "v${version}";
-            sha256 = "sha256-c6Feex0FGxxOWBEHxe0GqPHv65EwMMdxIeehZUGbl0Q=";
-          };
-        });
+      pkgs = import inputs.nixpkgs { system = system; overlays = [ ]; };
+      haskellPackages = pkgs.haskell.packages.ghc92.override {
+        overrides = self: super: {
+          riscv-hs = self.callCabal2nix "risv-hs" ./. { };
+        };
       };
-      pkgs = import inputs.nixpkgs { system = system; overlays = [ mob-overlay ]; };
-      haskellPackages = pkgs.haskell.packages.ghc92;
+      haskellDeps = drv: with builtins; concatLists (attrValues drv.getCabalDeps);
     in
     {
+      packages = pkgs // {
+        inherit haskellPackages;
+      };
       devShells.default = pkgs.mkShell {
         buildInputs = [
-          pkgs.mob
-          haskellPackages.ghc
+          (haskellPackages.ghcWithPackages (ps: haskellDeps ps.riscv-hs))
```

Comment:
I can't make a proper suggestion, because GitHub, but this is how I would rather write the `buildInput` part:
```
buildInputs = with haskellPackages; [
  (ghcWithPackages (ps: haskellDeps ps.riscv-hs))
  cabal-install
  haskell-language-server
];
```

This is more a matter of taste, and not a required change. I prefer this style here because it produces less noise in the list, and makes the long in-line function call easier on the eye.

Expanding the an attribute set such as `haskellPackages` with `with` is in general a double-edged sword for readability: If we maintained this file and added more packages from different attribute sets, e.g.
```
with foo; [ bar baz.bam ]
```
it would become hard to discern if e.g. `baz` is equal to `foo.baz` or comes from outside, such as `foo` did.

In this specific case I think it's slightly better to have less noise, and there is very little chance we will ever change these lines.

Reply:
Thanks. I missed that one.

Reply:
I see your point concerning readability. Let me implement this to get more practice with GitHub, even if it's just a matter of taste.